### ISSUE TITLE
master branch doesn't work in a web worker

### ISSUE THREAD

**Issue by GenaBitu** [showdownjs/showdown#537]
Hi, I switched to master (from the last npm version) to get #521, but showdown no longer works in a web worker, it errors with:
```
Uncaught ReferenceError: require is not defined
    at showdown.js:585
```
I include it in the web worker with:
```
importScripts('/node_modules/showdown/dist/showdown.js');
```

**Comment by tivie** [comment]
master is unstable. I will look into it as soon as possible.

**Comment by GenaBitu** [comment]
Thank you. I have already backported the fix to the latest release for my purposes, but it'd be nice for showdown to work in a Web worker in the future.

**Comment by tivie** [comment]
it will, of course. 

I actually cannot reproduce the issue you having. 

I'm behind schedule for v 2.0 ALPHA 1 release so I'm putting all my effort into it.

I will come back to this issue in the near future and will ask for your help in trying to reproduce it.

**Comment by GenaBitu** [comment]
Hi, I will look into it - maybe it's been fixed in the meantime, we'll see...

**Comment by GenaBitu** [comment]
Nope, the issue is still there. (line 587 now). I'll make a minimal example.

**Comment by GenaBitu** [comment]
Hey, I've created a [gist](https://gist.github.com/GenaBitu/5da0ed47609d39f43727291479ae94ca). The showdown file is the current master file `dist/showdown.js`. Bear in mind, that you can't run this code from a file, Web Workers only work when served from a server...

**Comment by GenaBitu** [comment]
From what I understand about modern JS (= not much...), line 587 is only supposed to run if you are in a Node environment, not in the browser. But the way this check is implemented, this check thinks it is in Node when in fact it's a Web Worker.

**Comment by tivie** [comment]
ah, it's the JSDOM shim that is breaking the webworker...
That code was provisory but that should be easy to fix.

**Comment by GenaBitu** [comment]
Yeah, I've tried commenting out lines 587 and 588, but then 590 breaks because Web workers don't have `this.window`, which I don't even know why this package would use :)

**Comment by tivie** [comment]
Because of the new feature of reverse coverting. We use the DOM parser to parse HTML into Markdown.

**Comment by GenaBitu** [comment]
Ah, I didn't know about that feature, thanks. That sounds really cool. A nit-picky question: Why did you decide to include this feature in this package as opposed to making it into its own one? I imagine that the reverse converting is a lot of code - making the package size needlessly bigger for people who don't use it (I would say the majority, but you know, I have a sample size of 1...).
Thanks!

**Comment by tivie** [comment]
Mainly for 2 reasons: They share a considerable portion of code and it's harder to maintain 2 libraries instead of one.

**Comment by andosteinmetz** [comment]
Hi, I just ran into this same issue at line `/node_modules/showdown/dist/showdown.js:2544` trying to use the HTML to markdown converter in Nodejs after updating to version 1.9.0. I had it working before updating, but I must have added a hack to fix it. My package.json previously had `"showdown" : "file:showdown"`.
